crates/interfaces/src/p2p/headers/client.rs

```diff
@@ -16,23 +17,29 @@ pub struct HeadersRequest {
     pub direction: HeadersDirection,
 }
 
+/// The headers future type
+pub type HeadersFuture<T> =
+    Pin<Box<dyn Future<Output = Result<PeerRequestResult<T>, RecvError>> + Send + Sync>>;
+
 /// The block headers downloader client
-#[async_trait]
 #[auto_impl::auto_impl(&, Arc, Box)]
 pub trait HeadersClient: DownloadClient {
+    /// The headers type
+    type Output;
+
```

Comment:
we already know what this type should return: `PeerRequestResult`

what we actually want is the Future type as the associated type like this:

```rust
pub trait HeadersClient {
    /// The headers type 
    type Output: Future<Output = PeerRequestResult<BlockHeaders>> ;

    /// Sends the header request to the p2p network and returns the header response received from a
    /// peer.
    fn get_headers(&self) -> Self::Output;
    
}

```

Reply:
That's what I tried at first but I hit a wall somewhere. I will give another try.

Reply:
if you're blocked please open a draft and we can discuss, it's always possible we missed something when creating the issue

Reply:
I don't know what I missed the first time, the update was pretty straightforward :/